examples/metrics/metrics/observer.js

```diff
@@ -14,29 +15,75 @@ const exporter = new PrometheusExporter(
 
 const meter = new MeterProvider({
   exporter,
-  interval: 2000,
+  interval: 3000,
 }).getMeter('example-observer');
 
-const otelCpuUsage = meter.createObserver('metric_observer', {
+meter.createObserver('cpu_core_usage', {
   monotonic: false,
-  labelKeys: ['pid', 'core'],
-  description: 'Example of a observer',
+  labelKeys: ['core'],
+  description: 'Example of a sync observer with callback',
+}, (observerResult) => {
+  observerResult.observe(getRandomValue(), { core: '1' });
+  observerResult.observe(getRandomValue(), { core: '2' });
 });
 
-function getCpuUsage() {
-  return Math.random();
-}
+const tempMetric = meter.createObserver('cpu_temp_per_app', {
+  monotonic: false,
+  labelKeys: ['app', 'core'],
+  description: 'Example of batch observer',
+});
+
+const cpuUsageMetric = meter.createObserver('cpu_usage_per_app', {
+  monotonic: false,
+  labelKeys: ['app', 'core'],
+  description: 'Example of batch observer',
+});
 
-const observable = new MetricObservable();
+meter.createBatchObserver('metric_batch_observer', (observerBatchResult) => {
+    function someAsyncMetrics() {
```

Comment:
I think this function should be defined outside the callback so it is clear it doesn't need to be defined within the callback.